dpnp/dpnp_iface_manipulation.py

```diff
@@ -1310,6 +1312,62 @@ def can_cast(from_, to, casting="safe"):
     return dpt.can_cast(dtype_from, to, casting=casting)
 
 
+def common_type(*arrays):
+    """
+    Return a scalar type which is common to the input arrays.
+
+    The return type will always be an inexact (i.e. floating point or complex)
+    scalar type, even if all the arrays are integer arrays.
+    If one of the inputs is an integer array, the minimum precision type
+    that is returned is determined by the device capabilities.
+
+    For full documentation refer to :obj:`numpy.common_type`.
+
+    Parameters
+    ----------
+    arrays: {dpnp.ndarray, usm_ndarray}
+        Input arrays.
+
+    Returns
+    -------
+    out: data type
+        Data type object.
+
+    Examples
+    --------
+    >>> import dpnp as np
+    >>> np.common_type(np.arange(2, dtype=np.float32))
+    numpy.float32
+    >>> np.common_type(np.arange(2, dtype=np.float32), np.arange(2))
+    numpy.float64
+    >>> np.common_type(np.arange(4), np.array([45, 6.j]), np.array([45.0]))
+    numpy.complex128
+
+    """
+
+    if len(arrays) == 0:
+        return (
+            dpnp.float16
+            if dpctl.select_default_device().has_aspect_fp16
+            else dpnp.float32
+        )
+
+    dpnp.check_supported_arrays_type(*arrays)
+
+    _, exec_q = get_usm_allocations(arrays)
+    default_float_dtype = dpnp.default_float_type(sycl_queue=exec_q)
+    dtypes = []
+    for a in arrays:
+        if a.dtype.kind == "b":
+            raise TypeError("can't get common type for non-numeric array")
+        if a.dtype.kind in "iu":
```

Comment:
It seems more clear to use explicit alias
```suggestion
        if dpnp.issubdtype(a.dtype, dpnp.integer):
```